File: Example/XliftSDKJavaExampleInfeed/app/src/main/java/jp/xlift/android/sdk/example/XliftInfeedAdViewModel.java
package jp.xlift.android.sdk.example;

import android.content.Intent;
import android.databinding.BindingAdapter;
import android.graphics.BitmapFactory;
import android.util.Log;
import android.view.View;
import android.widget.ImageView;

import java.io.InputStream;

import jp.xlift.android.sdk.infeed.XliftInfeedAd;

public class XliftInfeedAdViewModel {
    private XliftInfeedAd xliftInfeedAd;

    public XliftInfeedAdViewModel(XliftInfeedAd xliftInfeedAd) {
        this.xliftInfeedAd = xliftInfeedAd;
    }

    public XliftInfeedAd getXliftInfeedAd() {
        return xliftInfeedAd;
    }

    public String getTitle() {
        return this.xliftInfeedAd.getTitle();
    }

    public String getSourceTitle() {
        return xliftInfeedAd.getAdType() == XliftInfeedAd.AdType.Ad
                ? String.format("[PR] %s", xliftInfeedAd.getSiteTitle()) : xliftInfeedAd.getSiteTitle();
    }

    public void loadImage(final ImageView imageView) {
        this.xliftInfeedAd.loadImage(new XliftInfeedAd.LoadImageCallback() {
            @Override
            public void onSuccess(InputStream imageInputStream) {
                imageView.setImageBitmap(BitmapFactory.decodeStream(imageInputStream));
            }

            @Override
            public void onFailure(Exception e) {
                Log.w("XliftSDK-Example", e);
            }
        });
    }

    public void onClick(View view) {
        view.getContext().startActivity(
            new Intent(Intent.ACTION_VIEW, xliftInfeedAd.createClickUri())
        );
    }

    @BindingAdapter("setXliftImage")
    public static void setImage(final ImageView imageView, XliftInfeedAdViewModel viewModel) {
        viewModel.loadImage(imageView);
    }
}

File: Example/XliftSDKKotlinExampleInfeed/app/src/main/java/jp/xlift/android/sdk/example/XliftInfeedAdViewModel.kt
package jp.xlift.android.sdk.example

import android.content.Intent
import android.databinding.BindingAdapter
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.util.Log
import android.view.View
import android.widget.ImageView

import jp.xlift.android.sdk.infeed.XliftInfeedAd
import java.io.InputStream

class XliftInfeedAdViewModel(val xliftInfeedAd: XliftInfeedAd) {

    val title: String
        get() = this.xliftInfeedAd.title

    val sourceTitle: String
        get() = if (xliftInfeedAd.adType == XliftInfeedAd.AdType.Ad)
            String.format("[PR] %s", xliftInfeedAd.siteTitle)
        else
            xliftInfeedAd.siteTitle

    fun loadImage(imageView: ImageView) {
        this.xliftInfeedAd.loadImage(object : XliftInfeedAd.LoadImageCallback {
            override fun onSuccess(imageInputStream: InputStream) {
                imageView.setImageBitmap(BitmapFactory.decodeStream(imageInputStream))
            }

            override fun onFailure(e: Exception) {
                Log.w("XliftSDK-Example", e)
            }
        })
    }

    fun onClick(view: View) {
        view.context.startActivity(
            Intent(Intent.ACTION_VIEW, xliftInfeedAd.createClickUri())
        )
    }

    companion object {
        @BindingAdapter("setXliftImage")
        @JvmStatic
        fun setImage(imageView: ImageView, viewModel: XliftInfeedAdViewModel) {
            viewModel.loadImage(imageView)
        }
    }
}
